kernel/poll.c

```diff
@@ -197,6 +231,10 @@ int k_poll(struct k_poll_event *events, int num_events, s32_t timeout)
 	int last_registered = -1, rc;
 	unsigned int key;
 
+	if (num_events == 1) {
+		return poll_1(events, timeout);
```

Comment:
Have you benchmarked this?  The net effect here is a fairly significant increase in code size, and what you gain is AFAICT that the loop exit test in the needs to be executed only once (i.e. if(num_events == 1) is basically the same test) instead of twice, and the final irq_unlock/lock pair gets to be elided.

That's just a tiny handful of cycles.  I'm sure it's faster, but I doubt it's much faster and I'd say we really need to see numbers before adding this.

Reply:
My understanding is that this patch was an attempt to address https://jira.zephyrproject.org/browse/ZEP-2593 , but per my comment above, it doesn't help.

Reply:
Well, I was coming from the point that IRQ should be locked as little as possible since this is now used for by buffer polls this is a very heavy dependency on this code path.